wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift

```diff
@@ -569,13 +575,33 @@ final class ClientListViewController: UIViewController,
             self.clients = results
             self.selfClient = results.first(where: { $0.isSelfClient() })
             refreshViews()
+            completed?()
         }
     }
 
     @MainActor
     func refreshViews() {
         clientsTableView?.reloadData()
     }
+
+    private func updateE2EIdentityCertificateInDetailsView() {
+        let updateViewModel: (E2eIdentityCertificate?, String?) -> Void = {[weak self] certificate, mlsThumbprint in
+            self?.deviceInfoViewModel?.e2eIdentityCertificate = certificate
+            self?.deviceInfoViewModel?.mlsThumbprint = mlsThumbprint?.splitStringIntoLines(charactersPerLine: 16)
+        }
+
+        if deviceInfoViewModel?.isSelfClient == true {
+            updateViewModel(deviceInfoViewModel?.e2eIdentityCertificate,
+                            selfClient?.resolvedMLSThumbprint?.splitStringIntoLines(charactersPerLine: 16))
+        } else {
+            let client = clients.first(
+                where: {
+                $0.clientId == deviceInfoViewModel?.userClient.clientId
+            })
+            updateViewModel(deviceInfoViewModel?.e2eIdentityCertificate,
+                            client?.resolvedMLSThumbprint?.splitStringIntoLines(charactersPerLine: 16))
+        }
+    }
```

Comment:
I'm having difficulties understanding what's going on in this method and why it is needed.
Aren't your code and the following equivalent?

```suggestion
    private func updateE2EIdentityCertificateInDetailsView() {
        guard let deviceInfoViewModel else { return }

        let client = if deviceInfoViewModel.isSelfClient {
            selfClient
        } else {
            clients.first { $0.clientId == deviceInfoViewModel.userClient.clientId }
        }
        deviceInfoViewModel.mlsThumbprint = client?.resolvedMLSThumbprint?.splitStringIntoLines(charactersPerLine: 16)
    }
```

Reply:
`ClientListViewController` has list of clients that get updated with the certificate post enrolment or updating certificate. Though userclient instance is passed(pass by reference) to SwiftUI, somehow that instance does not get updated value. Hence, keeping a reference of the `DeviceInfoViewModel` here.

Reply:
rewrote this code